refactor: update replace spec (#11)
interlay/interbtc-spec: interbtc-spec/docs/source/spec/replace.rst

.. tabularcolumns:: |l|l|L|

======================  ===========  =======================================================	
Parameter               Type         Description                                            
======================  ===========  =======================================================
``oldVault``            AccountId    Account of the *oldVault* that is to be replaced.
``newVault``            AccountId    Account of the *newVault*, which accepts the replace request.
``amount``              interBTC     Amount of BTC / interBTC to be replaced.
``griefingCollateral``  DOT          Griefing protection collateral locked by *oldVault*.

> ``collateral``          DOT          DOT collateral locked by the new Vault.

Reviewer: ```suggestion
``collateral``          DOT          Collateral locked by the new Vault.
```

*Parameters*

* ``oldVault``: Account identifier of the Vault to be replaced (as tracked in ``Vaults`` in :ref:`vault-registry`).
* ``btcAmount``: Integer amount of BTC / interBTC to be replaced.
* ``griefingCollateral``: collateral locked by the *oldVault* as griefing protection.

*Events*

* :ref:`requestReplaceEvent`

*Preconditions*

* The function call MUST be signed by *oldVault*.
* The BTC Parachain status in the :ref:`security` component MUST be set to ``RUNNING:0``.
* The *oldVault* MUST be registered.

> * The *oldVault* MUST NOT be banned.

Reviewer: ```suggestion
* The *oldVault* MUST NOT be banned.
* The *oldVault* MUST NOT be nominated (if :ref:`vault_nomination` is enabled).
```

> * The *oldVault* MUST provide sufficient ``griefingCollateral`` such that the ratio of all of its ``toBeReplacedTokens`` and ``replaceCollateral`` is above :ref:`replaceGriefingCollateral`.

Reviewer: ```suggestion
* If the ``btcAmount`` is greater than the Vault's ``replacableTokens = issuedTokens - toBeRedeemTokens - toBeReplaceTokens``, set the ``btcAmount to the ``replaceableTokens`` amount.
* The *oldVault* MUST provide sufficient ``griefingCollateral`` such that the ratio of all of its ``toBeReplacedTokens`` and ``replaceCollateral`` is above :ref:`replaceGriefingCollateral`.
```

> * The *oldVault* MUST request sufficient tokens to be replaced such that its total is above ``ReplaceBtcDustValue``.

Reviewer: ```suggestion
* The *oldVault* MUST request sufficient ``btcAmount`` to be replaced such that its total is above ``ReplaceBtcDustValue``.
```

* The BTC Parachain status in the :ref:`security` component MUST NOT be set to ``SHUTDOWN: 2``.
* The *oldVault* and *newVault* MUST be registered.
* The *oldVault* MUST NOT be equal to *newVault*.
* The *newVault*'s free balance MUST be enough to lock ``collateral``.
* The *newVault* MUST have lock sufficient collateral to remain above the :ref:`SecureCollateralThreshold` after accepting ``btcAmount``.
* The *newVault*'s ``btcAddress`` MUST NOT be registered.
* The replaced tokens MUST be at least``ReplaceBtcDustValue``.

*Postconditions*

The actual amount of replaced tokens is calculated to be ``consumedTokens = min(oldVault.toBeReplacedTokens, btcAmount)``. The amount of griefingCollateral used is ``consumedGriefingCollateral = oldVault.replaceCollateral * (consumedTokens / oldVault.toBeReplacedTokens)``.

* The *oldVault*'s ``replaceCollateral`` MUST be decreased by ``consumedGriefingCollateral``. 
* The *oldVault*'s ``toBeReplacedTokens`` MUST be decreased by ``consumedTokens``. 
* The *oldVault*'s ``toBeRedeemedTokens`` MUST be increased by ``consumedTokens``. 
* The *newVault*'s ``toBeIssuedTokens`` MUST be increased by ``consumedTokens``. 

> * The *newVault* locks additional collateral; its ``backingCollateral`` MUST be increased by ``collateral * (consumedTokens / oldVault.toBeReplacedTokens)``. 

Reviewer: ```suggestion
* The *newVault* locks additional collateral; its ``backingCollateral`` MUST be increased by ``collateral * (consumedTokens / oldVault.toBeReplacedTokens)``. 
* A unique `replaceId` must be generated from :ref:`generateSecureId`.
```

> * A new ``ReplaceRequest`` MUST be added to storage. The amount is set to ``consumedTokens``, ``griefingCollateral`` to ``consumedGriefingCollateral``, ``collateral`` to the ``collateral`` argument, ``accept_time`` to the current active block number, ``period`` to the current ``ReplacePeriod``, ``btcAddress`` to the ``btcAddress`` argument, ``btc_height`` to the current height of the btc-relay, and ``status`` to ``pending``.

Reviewer: ```suggestion
* A new ``ReplaceRequest`` MUST be added to the replace request mapping under the `replaceId` key. 
  * ``oldVault``: MUST be the ``oldVault``.
  * ``newVault``: MUST be the ``newVault``.
  * ``amount``: MUST be``consumedTokens``. 
  * ``griefingCollateral``: MUST be ``consumedGriefingCollateral``
  * ``collateral``: MUST be ``collateral``.
  * ``accept_time``: MUST be the current active block number.
  * ``period``: MUST be the current ``ReplacePeriod``.
  * ``btcAddress``: MUST be the ``btcAddress`` argument.
  * ``btcHeight``: UST be the current height of the btc-relay.
  * ``status``: MUST be ``pending``.
```
